src/animators/animator.ts

```diff
@@ -1,4 +1,39 @@
 module Plottable {
+export module Animators {
+  export type d3SelectionOrTransition = d3.Selection<any>|d3.Transition<any>;
+
+  export type EasingFunction = (t: number) => number;
+  export type EasingFunctionSpecifier = string|EasingFunction;
+  export class EasingFunctions {
+    public static atStart = (t: number) => {
+      return 1;
+    };
+    public static atEnd = (t: number) => {
+      if (t < 1) {
+        return 0;
+      };
+      return 1;
+    };
+    public static squEase(easingFunction: EasingFunctionSpecifier, start: number, end: number): EasingFunction {
+      return (t: number) => {
+        if (start === undefined) {
+          start = 0;
+        };
+        if (t >= end) {
+            return 1;
+        }
+        if (t <= start) {
+          return 0;
+        }
+        let tbar = (t - start) / (end - start);
+        if (typeof (easingFunction) === "string") {
+          easingFunction = d3.ease(<string>easingFunction);
+        }
```

Comment:
interestingly, type gaurding doesn't apply in this case:

```javascript
easingFunction = d3.ease(easingFunction);
```
because the value of ```easingFunction``` gets modified

changing it to 
```javascript
return = d3.ease(easingFunction)(tbar);
```

does recognize ```easingFunction``` as a string

From the typescript language spec:

>> •	In the true expression of a conditional expression, the type of a variable or parameter is narrowed by a type guard in the condition when true, __provided the true expression contains no assignments to the variable or parameter__.